File: LeetCode 30 Day Challenge/04-moveZeroes.js
const moveZeroes = (nums) => {
    let first = 0
    let second = 0
    let numsLen = nums.length
    
    while (second < numsLen) {
        if (nums[first] === 0 && nums[second] !== 0) {
            [nums[first], nums[second]] = [nums[second], nums[first]]
        }
        if (nums[first] !== 0) {
            first++
        }
        second++
    }
}

File: LeetCode 30 Day Challenge/04-move_zeroes.py
class Solution:
    def moveZeroes(self, nums):
        i = 0
        n = len(nums)
        while i < n:
            if nums[i] == 0:
                nums.pop(i)
                nums.append(0)
                n -= 1
            else:
                i += 1